app/models/viewmodels/notificationspage/converters/ERSResponseConverter.scala

```diff
@@ -40,44 +40,44 @@ class ERSResponseConverter @Inject()(decoder: Decoder, dateTimeFormatter: DateTi
+      val roeCodeExplanation =
+        findDucrEntry(notification.entries).flatMap(_.roe).flatMap(buildRoeCodeExplanation).getOrElse(HtmlFormat.empty)
+      val soeCodeExplanation =
+        findDucrEntry(notification.entries).flatMap(_.soe).flatMap(buildSoeCodeExplanation).getOrElse(HtmlFormat.empty)
+      val icsCodeExplanation =
+        findDucrEntry(notification.entries).flatMap(_.ics).flatMap(buildIcsCodeExplanation).getOrElse(HtmlFormat.empty)
 
       NotificationsPageSingleElement(
         title = messages("notifications.elem.title.inventoryLinkingMovementTotalsResponse"),
-        timestampInfo = timestampInfoResponse(notification.timestampReceived),
-        content =
-          Html(roeCodeExplanation.getOrElse("") + soeCodeExplanation.getOrElse("") + icsCodeExplanation.getOrElse(""))
+        timestampInfo = dateTimeFormatter.format(notification.timestampReceived),
+        content = new Html(List(roeCodeExplanation, soeCodeExplanation, icsCodeExplanation))
       )
     } else {
       throw new IllegalArgumentException(s"Cannot build content for ${notification.responseType}")
     }
 
   private def findDucrEntry(entries: Seq[Entry]): Option[Entry] = entries.find(_.ucrType.contains("D"))
 
-  private def buildRoeCodeExplanation(roeCode: String)(implicit messages: Messages): Option[String] = {
+  private def buildRoeCodeExplanation(roeCode: String)(implicit messages: Messages): Option[Html] = {
     val RoeCodeHeader = messages("notifications.elem.content.inventoryLinkingMovementTotalsResponse.roe")
     val roeCodeExplanationText = decoder.roe(roeCode).map(code => messages(code.contentKey))
 
-    roeCodeExplanationText.map(explanation => paragraph(s"$RoeCodeHeader $explanation"))
+    roeCodeExplanationText.map(explanation => code_explanation(RoeCodeHeader, explanation))
   }
 
-  private def buildSoeCodeExplanation(soeCode: String)(implicit messages: Messages): Option[String] = {
+  private def buildSoeCodeExplanation(soeCode: String)(implicit messages: Messages): Option[Html] = {
     val SoeCodeHeader = messages("notifications.elem.content.inventoryLinkingMovementTotalsResponse.soe")
     val soeCodeExplanationText = decoder.ducrSoe(soeCode).map(code => messages(code.contentKey))
 
-    soeCodeExplanationText.map(explanation => paragraph(s"$SoeCodeHeader $explanation"))
+    soeCodeExplanationText.map(explanation => code_explanation(SoeCodeHeader, explanation))
   }
 
-  private def buildIcsCodeExplanation(icsCode: String)(implicit messages: Messages): Option[String] = {
+  private def buildIcsCodeExplanation(icsCode: String)(implicit messages: Messages): Option[Html] = {
     val icsCodeExplanationText = decoder.ics(icsCode).map(code => messages(code.contentKey))
 
-    icsCodeExplanationText.map(explanation => paragraph(explanation))
+    icsCodeExplanationText.map(explanation => paragraph(explanation)).map(Html(_))
```

Comment:
Maybe paragraph should return `Html` already ?

Reply:
👍